docs/0002-events-architecture.md

```diff
@@ -0,0 +1,84 @@
+# 2. Events Architecture
+
+Date: 2024-06-13
+
+## Status
+
+Proposed
+
+## Context
+
+Currently, in the Visual Editor, Google Analytics (GA) tracking is implemented by hardcoding event data attributes directly into the component. This approach has limitations:
```

Comment:
Maybe we can add some additional context even before this, such as what the problem statement? . e.g. "It is required for us to add GA4 tracking attributes to button clicks and select element changes on the toolbar. Some of the data exists within publishing apps (section page heading) and some of the data exists within Visual Editor (button text, selected text). GovUK Publishing Components have implemented functionality to track all button clicks on elements with data attributes attached, but there is nothing for Select element changes."  

Another thing we can add: should we also lay out the alternative options that we considered? (e.g. asking GOVUK publishing component to write new code to track select changes? and why we opted for this option instead)

Reply:
I've update it with your comments